core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2592,6 +2607,34 @@ class KafkaApis(val requestChannel: RequestChannel,
             response.groups.addAll(results)
           }
 
+          // Clients are not allowed to see topics that are not authorized for Describe.
+          val topicsToCheck = response.groups.stream()
```

Comment:
one small optimization - could we do this check only if `authorizer` is defined?